src/app/cli/src/services/odf_server/access_token_registry_service.rs

```diff
@@ -0,0 +1,308 @@
+        {
+            server_record.add_account_token(access_token);
+        } else {
+            let mut server_record = ServerAccessTokensRecord::new(
+                odf_server_frontend_url.clone(),
+                odf_server_backend_url.clone(),
+            );
+            server_record.add_account_token(access_token);
+
+            registry.push(server_record);
+        }
+
+        self.storage.write_access_tokens_registry(scope, &registry)
+    }
+
+    pub fn drop_access_token(
+        &self,
+        scope: AccessTokenStoreScope,
+        odf_server_frontend_url: &Url,
+    ) -> Result<(), InternalError> {
+        let account_name = self.account_name();
+
+        let registry_ptr = match scope {
+            AccessTokenStoreScope::User => &self.user_registry,
+            AccessTokenStoreScope::Workspace => &self.workspace_registry,
+        };
+
+        let mut registry = registry_ptr
+            .lock()
+            .expect("Could not lock access tokens registry");
+
+        if let Some(token_map) = registry
+            .iter_mut()
+            .find(|c| &c.frontend_url == odf_server_frontend_url)
+        {
+            if let Some(_) = token_map.drop_account_token(account_name) {
+                return self.storage.write_access_tokens_registry(scope, &registry);
+            }
+        }
+
+        Ok(())
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+impl kamu::domain::auth::OdfServerAccessTokenResolver for AccessTokenRegistryService {
+    fn resolve_odf_dataset_access_token(&self, odf_dataset_http_url: &Url) -> Option<String> {
+        let origin = odf_dataset_http_url.origin().unicode_serialization();
+        let odf_server_backend_url = Url::parse(origin.as_str()).unwrap();
+
+        if let Some(token_find_report) =
+            self.find_by_backend_url(AccessTokenStoreScope::Workspace, &odf_server_backend_url)
+        {
+            Some(token_find_report.access_token.access_token)
+        } else if let Some(token_find_report) =
+            self.find_by_backend_url(AccessTokenStoreScope::User, &odf_server_backend_url)
+        {
+            Some(token_find_report.access_token.access_token)
+        } else {
+            None
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+pub trait AccessTokenStore: Send + Sync {
+    fn read_access_tokens_registry(
+        &self,
+        scope: AccessTokenStoreScope,
+    ) -> Result<OdfServerAccessTokenRegistry, InternalError>;
+
+    fn write_access_tokens_registry(
+        &self,
+        scope: AccessTokenStoreScope,
+        registry: &OdfServerAccessTokenRegistry,
+    ) -> Result<(), InternalError>;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+const KAMU_TOKEN_STORE: &str = ".kamutokenstore";
```

Comment:
Not a fan of this file name. Makes me wonder if tokens will be the only thing we store there... 

Git uses very generic `.git-credentials` - perhaps we can name it similarly?

Reply:
I explicitly wanted to avoid using term "credentials", as this is not a correct terminology. By definition, "Credentials" is not about tokens, it's about data required before login, not what is obtained after:

> Credentials are a set of login or authentication data that verify a user's identity and grant them access to a particular system or service. Usually, credentials include a username and a password.

I.e. for AWS this would be valid, as it stores API keys, which is something used before login.

Looking at my own computer, I've found a few similar files that end with "rc", i.e. `.npmrc`, `.yarnrc`. This stands for "runtime configuration", although the term is debatable too.

Docker uses something close to what I've proposed: `.tokenseed`.

IPFS uses term `.keystore`.